test/tests/functional/pbs_qmgr.py

```diff
@@ -72,57 +72,56 @@ def test_listcmd_whitespaces(self):
+                qmgr_cmd = ["qmgr", "-c", "list sched"]
+            else:
+                qmgr_cmd = ["qmgr", "-c", "\'list sched\'"]
+            tempfd = os.fdopen(fd, "w")
             ret = self.du.run_cmd(self.server.hostname, qmgr_cmd,
                                   stdout=tempfd)
+
             self.assertTrue(ret['rc'] == 0)
             for line in tempfd:
-                self.check_whitespace_prefix(line)
+                self.__check_whitespace_prefix(line)
+            close(tempfd)
 
-        # Check 2: line extensions are prefixed with tabs and not spaces
-        # Create a random long, comma separated string
-        blah = "blah"
-        long_string = ""
-        for i in range(49):
-            long_string += blah + ","
-        long_string += blah
-        # Create a new vnode
-        attrs = {"resources_available.ncpus": 2}
-        vndef = self.mom.create_vnode_def("vn", attrs)
-        vndef_file = "tempdef"
-        self.mom.insert_vnode_def(vndef, fname=vndef_file)
-        # create_vnode_def added "[0]" to the vnode name passed to it
-        nodename = "vn[0]"
-        cmd = ["pbsnodes", "-av"]
-        self.du.run_cmd(cmd=cmd)
-        # Set 'comment' attribute to the long string we created above
-        attrs = {"comment": long_string}
-        self.server.manager(MGR_CMD_SET, VNODE, attrs, nodename)
-        # Execute "qmgr 'list node vn[0]'"
-        # The comment attribute should generate a line extension
-        if self.du.is_localhost(self.server.hostname) is True:
-            qmgr_cmd = ["qmgr", "-c", "list node " + nodename]
-        else:
-            qmgr_cmd = ["qmgr", "-c", "\'list node " + nodename + "\'"]
-        with open(fn, "w+") as tempfd:
+            # Check 2: line extensions are prefixed with tabs and not spaces
+            # Create a random long, comma separated string
+            blah = "blah"
+            long_string = ""
+            for i in range(49):
+                long_string += blah + ","
+            long_string += blah
+            # Create a new vnode
+            attrs = {ATTR_rescavail: 2}
+            vndef = self.mom.create_vnode_def("vn", attrs)
+            vndef_file = "tempdef"
+            self.mom.insert_vnode_def(vndef, fname=vndef_file)
+            # create_vnode_def added "[0]" to the vnode name passed to it
+            nodename = "vn[0]"
+            # Set 'comment' attribute to the long string we created above
+            attrs = {ATTR_comment: long_string}
+            self.server.manager(MGR_CMD_SET, VNODE, attrs, nodename)
+            # Execute "qmgr 'list node vn[0]'"
+            # The comment attribute should generate a line extension
+            if self.du.is_localhost(self.server.hostname) is True:
+                qmgr_cmd = ["qmgr", "-c", "list node " + nodename]
+            else:
+                qmgr_cmd = ["qmgr", "-c", "\'list node " + nodename + "\'"]
+            tempfd = open(fn, "w+")
```

Comment:
just curious why you are opening file in w+ mode here but in write mode in the previous open call.